src/Processors/Formats/Impl/Parquet/Write.cpp

```diff
@@ -396,6 +398,43 @@ struct ConverterNumberAsFixedString
     size_t fixedStringSize() { return sizeof(T); }
 };
 
+struct ConverterJSON
+{
+    using Statistics = StatisticsStringRef;
+
+    const ColumnObject & column;
+    PODArray<parquet::ByteArray> buf;
+    std::vector<String> stash;
+
+    explicit ConverterJSON(const ColumnPtr & c)
+        : column(assert_cast<const ColumnObject &>(*c))
+    {
+    }
+
+    const parquet::ByteArray * getBatch(size_t offset, size_t count)
+    {
+        buf.resize(count);
+        stash.clear();
+        stash.reserve(count);
+
+        auto data_type = std::make_shared<DataTypeObject>(DataTypeObject::SchemaFormat::JSON);
```

Comment:
We cannot use default JSON data type here. We have to propogate the original data type of the JSON column, because it may have some parameters like `JSON(some.path SomeType)` that can impact the parsing